source/vir/src/ast_to_sst.rs

```diff
@@ -943,18 +943,34 @@ fn if_to_stm(
     }
 }
 
+fn expr_ret_never(expr: &Expr) -> bool {
+    if let TypX::Decorate(crate::ast::TypDecoration::Never, ..) = expr.typ.as_ref() {
+        true
+    } else {
+        false
+    }
+}
+
 /// Convert a VIR Expr to a SST (Vec<Stm>, ReturnValue), i.e., instructions of the form,
 /// "run these statements, then return this side-effect-free expression".
 ///
 /// Note the 'ReturnValue' can be one of three things:
 ///  * Some(e) - means the expression e was returned
 ///  * Unit - for the implicit unit case
 ///  * Never - the expression can never return (e.g., after a return value or break)
-
 pub(crate) fn expr_to_stm_opt(
     ctx: &Ctx,
     state: &mut State,
     expr: &Expr,
+) -> Result<(Vec<Stm>, ReturnValue), VirErr> {
```

Comment:
unfortunately, it's not going to be correct to do this, because the never type needs to be considered differently in Spec code. Consider:

```
let x = arbitrary::<!>();
assert(false);
```

The reason the Expr::NeverToAny is allowed to return ReturnValue::Never is because we check (in modes.rs) that NeverToAny is never used in Spec code.

Reply:
Thanks for the example. I was not aware of the difference in Spec code. I restricted the NeverReturn check in ExprX::Call when it is in exec mode.